components-java-sdk/src/main/java/io/dapr/components/domain/state/QueryItem.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.components.domain.state;
+
+import com.google.protobuf.ByteString;
+
+/**
+ * QueryItem is an object representing a single entry in query results.
+ *
+ * @param key The returned item Key.
+ * @param data The returned item Data.
+ * @param etag The returned item ETag.
+ * @param error The returned error string.
+ * @param contentType  The returned content-type.
+ */
+public record QueryItem(String key, ByteString data, String etag, String error, String contentType) {
```

Comment:
Side questions: How about renaming this to QueryResponseItem. I was re-reading the code and good thing I added a comment, because I could not remember what this class was for - requests or responses.

Reply:
I think that'd be perfectly ok, though I'm not sure I'd say it's necessary.  Any modern IDE would be able to quickly show references to it that made it apparent as to whether it's part of a request or a response.